src/cpp/core/system/Win32OutputCapture.cpp

```diff
@@ -44,13 +44,22 @@ void standardStreamCaptureThread(
       {
          const int kBufferSize = 512;
          char buffer[kBufferSize];
-         DWORD bytesRead = 0;
-         if (::ReadFile(hReadPipe, &buffer, kBufferSize, &bytesRead, NULL))
+         // read from the descriptor; this descriptor is attached to a pipe,
+         // and this _read call blocks until we have some bytes or until the
+         // descriptor is closed
+         DWORD bytesRead = ::_read(readFd, &buffer, kBufferSize);
+         if (bytesRead > 0)
          {
-            if (bytesRead > 0)
-               outputHandler(std::string(buffer, bytesRead));
+            // if we got some bytes, invoke the output handler
+            outputHandler(std::string(buffer, bytesRead));
          }
-         else
+         else if (bytesRead == 0)
```

Comment:
It doesn't look like this typically happens even when the process ends so adding a warning is appropriate. Updated commit here:

https://github.com/rstudio/rstudio/commit/d2aa97ad903a04fdf1f77fd2c1e726c468a3ef14#diff-cf87fcd202a44598c8656c66aeff39baR60